src/pages/index.js

```diff
@@ -18,80 +19,205 @@ import {
+
+const apiURL = {
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
   headers: {
     authorization: "a1101938-3641-4790-a37b-6b7f03e0e338",
     "Content-Type": "application/json",
   },
-});
+};
 
-fetch("https://around.nomoreparties.co/v1/cohort-3-en/users/me", {
-  method: "PATCH",
-  headers: {
-    authorization: "c56e30dc-2883-4270-a59e-b2f7bae969c6",
-    // "Content-Type" "application/json"
+// Add Card Popup//
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  (cardData) => {
+    addCardPopup.renderLoading();
+    api
+      .addNewCard(cardData)
+      .then((newCardData) => {
+        renderCard(newCardData);
+        addCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        addCardPopup.hideLoading();
+      });
   },
-  body: JSON.stringify({
-    name: "Marie Skłodowska Curie",
-    about: "Physicist and Chemist",
-  }),
-});
-// Card
-function handleCardImageClick(cardData) {
-  imagePreviewModal.open(cardData);
-}
+  "Saving..."
+);
 
-function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardImageClick);
-  const cardElement = card.getView();
-  section.addItem(cardElement);
+// Delete Card //
+const deleteCardPopup = new PopupWithConfirmation(
+  "#delete-card-modal",
+  "Deleting..."
+);
+deleteCardPopup.setEventListeners();
+function handleDeleteClick(card, cardID) {
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.showLoading();
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.hideLoading();
+      });
+  });
+
+  deleteCardPopup.open();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+// Editing Profile //
+const profileAvatar = document.querySelector(".profile__image");
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (inputValues) => {
+    editAvatarPopup.renderLoading();
+
+    const avatarData = {
+      avatar: inputValues.link,
+    };
+
+    api
+      .editProfileImage(avatarData)
+      .then(() => {
+        profileAvatar.src = inputValues.link;
+        editAvatarPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        editAvatarPopup.hideLoading();
+      });
   },
-  containerSelector
+  "Saving..."
 );
 
-section.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-// Modal Image
-const imagePreviewModal = new PopupWithImage(imageModalSelector);
-imagePreviewModal.setEventListeners();
+editAvatarPopup.setEventListeners();
+
+// Likes///
+function handleLikeClick(card) {
+  if (card.isLiked()) {
+    api
+      .likeCountRemove(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
```

Comment:
Please implement this method in the Card class, like in my snippet in Card.js file